### ISSUE TITLE
MW4 is slewing mount while image is still being exposed.

### ISSUE THREAD

**Issue by tbessell** [mworion/MountWizzard4#176]
I'm using Indi/Ekos on a Raspberry Pi 4. While MW4 is building the model, about half the images contain a light streak on the brightest stars. This does not prevent astrometry.net from solving the image, but I wonder if it affecting the accuracy. I have adjusted the mount settle time, but this makes no difference because the problem occurs at the end of image exposure and the next mount slew.
[image-012.fits.zip](https://github.com/mworion/MountWizzard4/files/7087555/image-012.fits.zip)

**Comment by mworion** [comment]
Hi,
sorry to hear this. This might be an issue with the INDI states transfer as getting the download mode right is tricky. It would be very helpful, if you could post the LOG file ideally in trace mode (under settings/misc). If you did not have the log in trace mode, please set it and make another test run. In this case it is not necessary to get real images nor they have to be solveable. 
Many thanks,
Michel

**Comment by tbessell** [comment]
Michel,
Thank you for a fine piece of software and for the suggestion. I will try to get a log file tonight.

I should mention that I am not running MW4 in a python virtual environment, but rather from a local installation. Is it possible that could be the cause?

**Comment by mworion** [comment]
No, MW4 will install anyway a virtual environment through the scripts. It's definitely better as running plain as the config of the systems and their installed libraries might interfere.

**Comment by tbessell** [comment]
hahaha! Well, I'm running a really customized system and I didn't use the install scripts, but one thing at a time. I will post a log file soon as I am able.
Thank you again for looking into this, it is greatly appreciated.
Tim

**Comment by tbessell** [comment]
Michel,
Here is a log file for your inspection. I had some problems that you might see at the beginning of the log. I didn't notice that my camera was not connecting in mw4. After a few tries, I was able to run a few models. The log should contain all the info.
Thank you kindly for your tremendous effort and clear skies to you,
Tim
[mw4-2021-09-02.log.zip](https://github.com/mworion/MountWizzard4/files/7100708/mw4-2021-09-02.log.zip)

**Comment by mworion** [comment]
Hi Tim,
I think I understood (hopefully) the situation. Just to explain: I use a signal after image is integrated and starting to download as the start signal for slewing the mount to the next point. ASCOM and ALPACA could provide these signals, INDI does not. So to make this happen there is a workaround I made: I wait for exposure = 0 as end of integration as replacement. When I look to the INDi driver implementation, the send of exposure = 0 comes before the driver starts to grab the image from the camera. So it may be the case on a raspi that the performance is sometimes not good enough to process the grab before sending the image. My suspicion is, that starting the grab also ends the image integration in the ASI framework. If so I only could think about workarounds.
There are two ways to do this: First: hanging the ASI INDI driver -> has to be done by someone else as I'm not the owner of the driver. Second: Introducing a conservative timing mode for MW4. This could be done by myself, but takes some time.

But anyway: if the images with the single trails look like the on you posted, there should be no impact on solving as the solver normally throws start rails or undefined clusters away. And you ave more than enough other stars for solving.

I'll keep this point open for the timing issue and try to provide a beta with this function for you to test.
Michel

**Comment by tbessell** [comment]
Michel,
After looking at the trace log, your explanation is about what I surmise is the problem, namely timing. I also looked at some of the code in mw4 (the trace log hints), but I don't know python very well. MW4 quite complex, thought I understand the logic enough to follow code.

One additional point. It seems that the star trails happen mostly on the longer slews. When the mount only has to move a relatively short distance, the images are perfect.

All that aside, I will trust your conclusion that it has no effect on plate solving and wait for a beta to test for you.

Thanks again for looking into this. Hopefully I can help make MountWizzard a better product for the astronomy community.
Tim

**Comment by mworion** [comment]
Tim,
important hint about the slew time in relation to the star trails. My explanation bases on an to early slew start, you is pointing to a too early exposing start. The latter one show be addressable by mount settle time. As I have a very rigid system with a focal length only 570mm I could set the settle time to 0 without seeing any issues. I know people with fl of about 2000mm - 4000mm which use settle times of 20-30 s to get their system set. To distinguish between this two assumptions: you said mostly. To be precise about the correlation: does mostly mean you never saw a short slew with trails ? Or does is say whenever I saw trails it was a longer slew ?
Yes, all feedback helps to make it better as the systems (computer setup, used framework and astronomy setup) vary that much and mostly I don't have or know the components used.
Michel

**Comment by tbessell** [comment]
Michel,
Now I wonder if my complaint suggests a wrong conclusion. I tried mount settle times up to 12 seconds with no difference in the images, maybe that wasn't enough. Next opportunity I have I will try to discover the condition which causes the trails more precisely.

Mount GM1000 HPS, telescope 10" RC fl 2000mm

thanX
Tim

**Comment by tbessell** [comment]
Michel,
Now I really feel kind of stupid because I bothered you with this non-existent issue.
Your last post that suggested mount settle time was the real answer to my problem. I tried settle times of 30 s and 20 s and that seems to have fixed the problem. No more star trails!

I ran three models, after the first and second, I did a polar alignment. Then ran a third model, RMS 3.6" without deleting a single point.

Thank you so, so much for helping me understand and again, sorry I made you chase a problem that doesn't exist. New mount, new scope, and new software. Really through me off!

Tim
